src/frame.h

```diff
@@ -69,6 +70,10 @@ struct frame_additional_data : frame_header
 
     uint32_t raw_size = 0;     // The frame transmitted size (payload only)
 
+    // Used by software-device, to save metadata that doesn't have enough room in metadata_blob
+    // (saving sizeof(rs2_frame_metadata_value) + sizeof(rs2_metadata_type) for each value)
+    std::vector< std::pair< rs2_frame_metadata_value, rs2_metadata_type > > sw_device_extra_data;
```

Comment:
So this is basically a mapping from type -> value, yes? (based on `try_get`)
Why a vector?

Reply:
You are right. We can replace it with a map.